File: wagerproof-ios-native/Wagerproof/Features/Onboarding/OnboardingPageSlot.swift
import SwiftUI

extension EnvironmentValues {
    /// True while a page is the active carousel page. With the button-driven
    /// slide pager only the active page is mounted, so this defaults to true
    /// — pages gate their Lottie playback / chart grow-ins / entrances on it
    /// (`.pageEntrance(index:)`, the ATT prompt) and the trigger fires at
    /// mount, which is exactly when the page starts sliding in.
    @Entry var onboardingPageIsActive: Bool = true
}

File: wagerproof-android-native/app/src/main/java/com/wagerproof/app/features/onboarding/OnboardingPageSlot.kt
package com.wagerproof.app.features.onboarding

import androidx.compose.runtime.staticCompositionLocalOf

/**
 * Port of iOS `OnboardingPageSlot.swift`'s `\.onboardingPageIsActive`
 * environment entry. True while a page is the active carousel page. With the
 * button-driven slide pager only the active page is mounted, so this defaults
 * to true — pages gate their entrance choreography (`pageEntrance(index:)`) and
 * one-shot animations on it, and the trigger fires at mount, which is exactly
 * when the page starts sliding in.
 */
val LocalOnboardingPageIsActive = staticCompositionLocalOf { true }
